controller.go

```diff
@@ -337,6 +341,11 @@ func (c *controller) clusterAgentInit() {
 			// service bindings
 			c.agentClose()
 			c.cleanupServiceBindings("")
+
+			c.Lock()
+			close(c.agentStopDone)
```

Comment:
This assumes that `c.agentStopDone` is non-nil. How do we know that `agentSetup` was called already? That's another case statement above.